openmdao/visualization/inputs_report/inputs_report.py

```diff
@@ -0,0 +1,150 @@
+from openmdao.utils.reports_system import register_report
+from openmdao.visualization.tables.table_builder import generate_table
+
+
+def _unit_str(meta):
+    try:
+        u = meta['units']
+    except KeyError:
+        return ''
+    return '' if u is None else u
+
+
+def _get_val_cells(val):
+    if isinstance(val, np.ndarray):
+        minval = np.min(val)
+        maxval = np.max(val)
+        if val.size > 5:
+            val = f"| {np.linalg.norm(val)} |"
+        else:
+            val = np.array2string(val)
+        return val, minval, maxval
+
+    return val, None, None
+
+
+def inputs_report(prob, outfile=None, display=True, precision=6, title=None,
+                  tablefmt='tabulator'):
+    """
+    Generate a self-contained html file containing a detailed inputs report.
+
+    Optionally pops up a web browser to view the file.
+
+    Parameters
+    ----------
+    prob : Problem
+        View the inputs for this Problem.
+    outfile : str, optional
+        The name of the output file.  Defaults to 'inputs.html' for web-based
+        table formats and stdout for text based ones.
+    display : bool, optional
+        If True, display the table. Defaults to True.
+    precision : int, optional
+        Sets the precision for displaying array values.
+    title : str, optional
+        Sets the title of the web page.
+    tablefmt : str, optional
+        Format for generated table. Should be one of ['text', 'github', 'rst', 'html', 'tabulator'].
+        Defaults to 'tabulator' which generates a sortable, filterable web-based table.
+    """
+    if isinstance(prob, Problem):
+        model = prob.model
+    else:
+        raise RuntimeError("Input report requires a Problem instance, but got a "
+                           f"'{type(prob).__name__}'.")
+
+    connections = model._conn_global_abs_in2out
+
+    if not connections:  # only possible if top level system is a component
+        return
+
+    if model._outputs is None:
+        raise RuntimeError("Can't generate inputs report. Input values are unknown because "
+                           "final_setup has not been called.")
+
+    # get absolute src names of design vars
+    desvars = model.get_design_vars(recurse=True, use_prom_ivc=False)
+
+    rows = []
+    with printoptions(precision=precision, suppress=True, threshold=10000):
+        for target, meta in model._var_allprocs_abs2meta['input'].items():
+            prom = model._var_allprocs_abs2prom['input'][target]
+            src = connections[target]
+            val = model.get_val(target, get_remote=True, from_src=not src.startswith('_auto_ivc.'))
+            smeta = model._var_allprocs_abs2meta['output'][src]
+            src_is_ivc = 'indep_var' in smeta['tags']
+            vcell, mincell, maxcell = _get_val_cells(val)
+
+            rows.append([target, prom, src, src_is_ivc, src in desvars, _unit_str(meta),
+                         meta['shape'], sorted(meta['tags']), vcell, mincell, maxcell])
```

Comment:
Possibly round these values to or change the way they are displayed in the table so that they don't show 16 digits of accuracy.  Seems like 6 digits with an exponential format would be plenty and keep the cells from wrapping.

Reply:
I changed it to a {:.6g} format.  Let me know if that's any better.